src/main/java/co/rsk/federate/signing/hsm/message/ReleaseCreationInformation.java

```diff
@@ -48,6 +58,29 @@ public ReleaseCreationInformation(
         this.pegoutCreationRskTxHash = pegoutCreationRskTxHash;
         this.pegoutBtcTx = pegoutBtcTx;
         this.pegoutConfirmationRskTxHash = pegoutConfirmationRskTxHash;
+
+        this.decodeUtxoOutpointValues(transactionReceipt);
+    }
+
+    private void decodeUtxoOutpointValues(TransactionReceipt transactionReceipt) {
+        List<LogInfo> logs =  transactionReceipt.getLogInfoList();
+
+        Function pegoutTransactionCreatedEvent = BridgeEvents.PEGOUT_TRANSACTION_CREATED.getEvent();
+        final byte[] pegoutTransactionCreatedSignatureTopic = pegoutTransactionCreatedEvent.encodeSignatureLong();
+
+        Optional<LogInfo> pegoutTransactionCreatedLog = logs.stream().filter((log ->
+            !log.getTopics().isEmpty() && Arrays.equals(log.getTopics().get(0).getData(),
+                pegoutTransactionCreatedSignatureTopic)
+        )).findFirst();
+
+        if(!pegoutTransactionCreatedLog.isPresent()) {
+            return;
+        }
+
+        byte[] pegoutCreatedTransactionEventData = pegoutTransactionCreatedLog.get().getData();
+        byte[] serializedOutpointValues = (byte[]) pegoutTransactionCreatedEvent.decodeEventData(pegoutCreatedTransactionEventData)[0];
+
+        this.utxoOutpointValues = Collections.unmodifiableList(UtxoUtils.decodeOutpointValues(serializedOutpointValues));
```

Comment:
I was thinking about this and since we are using the Optional API we can make this more functional. Something like this, wdyt?

```java
private void decodeUtxoOutpointValues(TransactionReceipt transactionReceipt) {
   return transactionReceipt.getLogInfoList().stream()
        .filter(this::isPegoutTransactionCreatedLog)
        .findFirst()
        .map(LogInfo::getData)
        .map(this::decodePegoutTransactionEventData)
        .map(UtxoUtils::decodeOutpointValues)
        .map(Collections::unmodifiableList)
        .orElse(Collections.emptyList());
}

private boolean isPegoutTransactionCreatedLog(LogInfo log) {
    Function pegoutTransactionCreatedEvent = BridgeEvents.PEGOUT_TRANSACTION_CREATED.getEvent();
    byte[] pegoutTransactionCreatedSignatureTopic = pegoutTransactionCreatedEvent.encodeSignatureLong();

    return !log.getTopics().isEmpty() && 
           Arrays.equals(log.getTopics().get(0).getData(), pegoutTransactionCreatedSignatureTopic);
}

private byte[] decodePegoutTransactionEventData(byte[] eventData) {
    return (byte[]) pegoutTransactionCreatedEvent.decodeEventData(eventData)[0];
}
```

Reply:
I very like your proposal.

Reply:
Done